recipes/scons/all/conanfile.py

```diff
@@ -0,0 +1,69 @@
+from conans import ConanFile, tools
+from conans.errors import ConanException
+import os
+import shutil
+import sys
+
+
+class SConsConan(ConanFile):
+    name = "scons"
+    description = "SCons is an Open Source software construction tool—that is, a next-generation build tool"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index/"
+    homepage = "https://scons.org"
+    topics = ("conan", "scons", "build", "configuration", "development")
+    settings = "os_build"  # Added to test on let CI test on all os'es
+
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("scons-{}".format(self.version), self._source_subfolder)
+
+    def build(self):
+        with tools.chdir(self._source_subfolder):
+            self.run("{} setup.py build -j{}".format(sys.executable, tools.cpu_count()))
+
+    def package(self):
+        self.copy("LICENSE*", src=self._source_subfolder, dst="licenses")
+
+        # Mislead CI and create an empty header in the includue directory
+        include_dir = os.path.join(self.package_folder, "include")
+        os.mkdir(include_dir)
+        tools.save(os.path.join(include_dir, "__nop.h"), "")
+
+        with tools.chdir(self._source_subfolder):
+            self.run("{} setup.py install --no-compile --prefix={}".format(sys.executable, self.package_folder))
+
+        tools.rmdir(os.path.join(self.package_folder, "man"))
+
+        if tools.os_info.is_windows:
+            shutil.move(os.path.join(self.package_folder, "Scripts"),
+                        os.path.join(self.package_folder, "bin"))
+            shutil.move(os.path.join(self.package_folder, "Lib"),
+                        os.path.join(self.package_folder, "lib2"))
+            shutil.move(os.path.join(self.package_folder, "lib2"),
+                        os.path.join(self.package_folder, "lib"))
+
+        # Check for compiled python sources
+        for root, _, files in os.walk(self.package_folder):
```

Comment:
why this is needed?

Reply:
I think it should be moved to the test package.

Reply:
Yeah, good idea.

Reply:
I remember why I put it in `package`:
python automatically byte compiles modules as it uses them.
If I do this in the `package` method of the recipe itself, no such objects will be in the folder.
But this test is done in the test_package, a previous run of scons will cause this test to fail.
Running `conan test test_package scons/3.1.2@ -tbf /tmp/scons_tp` twice will not work then.

Reply:
maybe it's better to remove that files, instead of failing the build?
also, I suppose package method support filtering (** excludes**)

Reply:
`self.copy` has an *excludes* option.
The recipe is not using `self.copy` but install of `setup.py`.
But I'll change it to: remove the `.py[cdow]` and print a warning.

Reply:
yeah, much better

Reply:
Is there something left to do with this?

Reply:
I don't think so. @SSE4 seems to be happy 🤣 .
I'm resolving this 😄 .